lib/boorus/moebooru/feats/favorites/moebooru_favorites_provider.dart

```diff
@@ -1,10 +1,68 @@
 // Package imports:
+import 'package:flutter/widgets.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
 // Project imports:
+import '../../../../core/configs/ref.dart';
+import '../../../../core/configs/src/create/providers.dart';
+import '../../../../core/configs/src/manage/current_booru_providers.dart';
+import '../../../../core/posts/favorites/widgets.dart';
+import '../../../../core/posts/post/src/types/post.dart';
+import '../../moebooru.dart';
 import 'moebooru_favorites_notifier.dart';
 
 final moebooruFavoritesProvider =
     NotifierProvider.family<MoebooruFavoritesNotifier, Set<String>?, int>(
   MoebooruFavoritesNotifier.new,
 );
+
+class MoebooruQuickFavoriteButton extends ConsumerWidget {
+  const MoebooruQuickFavoriteButton({
+    required this.post,
+    super.key,
+  });
+
+  final Post post;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final config = ref.watchConfigAuth;
+    final notifier = ref.watch(moebooruFavoritesProvider(post.id).notifier);
+    final canFavorite = config.hasLoginDetails();
+    final hasData = ref.watch(moebooruFavoritesProvider(post.id)) != null;
+    final forceShowFavoriteStatus = ref.watch(
```

Comment:
Should check null for this setting first. If it is null (default behavior) then we will return a SizedBox immediately without doing anything.